src/ZkFold/Symbolic/Data/ByteString.hs

```diff
@@ -86,15 +331,104 @@ instance (Arithmetic a, KnownNat n) => Arithmetizable a (ByteString n (Arithmeti
 
     typeSize = minNumberOfRegisters @a @n
 
-binOp
-    :: forall a n .
-       Arithmetic a
+
+-- TODO: I really don't like that summation is implemented here and in UInt. Can we do something about it?
+-- Converting ByteStrings to UInt and back will flood the circuit with new constraints because of different bit layouts in these types.
```

Comment:
Well, this is actually not a problem because
1. The cast needed in SHA-2 algorithms is between `ByteString 32` and `UInt 32` (or between `ByteString 64` and `UInt 64`), which is a no-op in the fields we use because they both need a single register
2. If someone needs a cast for big `n`, this would not be a problem anyway once we add range checks